src/System.Security.Permissions/src/System/Xaml/Permissions/XamlAccessLevel.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Security;
+
+namespace System.Xaml.Permissions
+{
+    /// <SecurityNote>
+    /// This class is immutable. Various consumers copy references into SecurityCritical fields,
+    /// and their security depends on the immutability of the members defined here.
+    /// Derived classes may add mutable members, those have no impact on the consumers of this class.
+    /// </SecurityNote>
+    [Serializable]
+    public class XamlAccessLevel
+    {
+        private XamlAccessLevel(string assemblyName, string typeName)
+        {
+            AssemblyNameString = assemblyName;
+            PrivateAccessToTypeName = typeName;
+        }
+
+        public static XamlAccessLevel AssemblyAccessTo(Assembly assembly)
+        {
+            if (assembly == null)
+            {
+                throw new ArgumentNullException(nameof(assembly));
+            }
+            return new XamlAccessLevel(assembly.FullName, null);
+        }
+
+        public static XamlAccessLevel AssemblyAccessTo(AssemblyName assemblyName)
+        {
+            if (assemblyName == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyName));
+            }
+            ValidateAssemblyName(assemblyName, "assemblyName");
+            return new XamlAccessLevel(assemblyName.FullName, null);
+        }
+
+        public static XamlAccessLevel PrivateAccessTo(Type type)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+            return new XamlAccessLevel(type.Assembly.FullName, type.FullName);
+        }
+
+        public static XamlAccessLevel PrivateAccessTo(string assemblyQualifiedTypeName)
+        {
+            if (assemblyQualifiedTypeName == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyQualifiedTypeName));
+            }
+            int nameBoundary = assemblyQualifiedTypeName.IndexOf(',');
+            if (nameBoundary < 0)
+            {
+                throw new ArgumentException("", nameof(assemblyQualifiedTypeName));
+            }
+            
+            string typeName = assemblyQualifiedTypeName.Substring(0, nameBoundary).Trim();
+            string assemblyFullName = assemblyQualifiedTypeName.Substring(nameBoundary + 1).Trim();
+            AssemblyName assemblyName = new AssemblyName(assemblyFullName);
+            ValidateAssemblyName(assemblyName, "assemblyQualifiedTypeName");
+            
+            return new XamlAccessLevel(assemblyName.FullName, typeName);
+        }
+
+        // Read-only: these properties should not be allowed to be modified once this object
```

Comment:
What do these comments refer to? Convert these standalone comments to doc-comments and attach them to the corresponding property.

Reply:
Deleted comment as part of hollowing out work